docs/migrating-to-v7.md

```diff
@@ -0,0 +1,127 @@
+# Migrating to v7
+
+First of all, don't worry - this is nowhere near as big a change as v6!
+
+Thanks to all the feedback we got from the community, we wanted to make some changes to our library to remove friction introduced with v6.
+There were no major refactors in this version, but the library's public API changed in some places.
+
+## Table of contents
+
+To make it easier to navigate this guide, here is an overview of the sections it contains:
+
+- [Thread-safe REST resource pagination](#thread-safe-rest-resource-pagination)
+- [Synchronous logger calls](#synchronous-logger-calls)
+- [`shopify.webhook.addHandlers` is no longer `async`](#shopifywebhookaddhandlers-is-no-longer-async)
+- [Removing deprecated code](#removing-deprecated-code)
+
+---
+
+## Thread-safe REST resource pagination
+
+The library now returns pagination info as part of `.all()` requests in REST resources, and no longer uses the `[PREV|NEXT]_PAGE_INFO` static, thread unsafe attributes.
+Instead of returning a plain array of objects, it will now return an object containing that array, as well as the response headers and pagination info.
```

Comment:
```suggestion
Instead of returning a plain array of objects, it will now return an object containing that array (in a `data` property), as well as the response headers and pagination info.
```